m2m-transfers/V3/REPWRITERSPECS/BANNO.M2MTRANSFERS.RPT.BATCH

```diff
@@ -0,0 +1,751 @@
+ IF TARGETENDDATE=DATENULL THEN
+  TARGETENDDATE=SYSTEMDATE
+
+ IF TARGETSYMXUSER<800 OR
+    TARGETSYMXUSER>899 THEN
+  DO
+   PRINT FORMAT("Invalid SymXchange user number:999 - TERMINATING PROGRAM RUN.",TARGETSYMXUSER)
+   NEWLINE
+   TERMINATE
+  END
+ ELSE IF TARGETENDDATE<TARGETSTARTDATE OR
+         TARGETSTARTDATE>SYSTEMDATE THEN
+  DO
+   PRINT "Invalid start/end date(s) - TERMINATING PROGRAM RUN."
+   NEWLINE
+   TERMINATE
+  END
+END [PROCEDURE]
+
+PROCEDURE FETCHLETTERFILEPARAMS
+[* Open Letter file BANNO.M2MTRANSFERS.V3.CFG to retrieve needed
+** parameter settings: 
+** 'TC:' (tran comment to search for)
+** 'XA:' (cross accounts allowed flag
+*]
+ XACCTACCESS=FALSE
+
+ FILEOPEN("LETTER",CFGFILENAME,"READ",CFGFILENUMBER,CFGFILEERROR)
+ IF CFGFILEERROR="" THEN
+  DO
+   WHILE (CFGFILEERROR="" AND
+          TARGETCOMMENT="")
+    DO
+     FILEREADLINE(CFGFILENUMBER,CFGFILELINE,CFGFILEERROR)
+     IF CFGFILEERROR="" THEN
+      DO
+       IF TARGETCOMMENT="" AND
+          UPPERCASE(SEGMENT(CFGFILELINE,1,3))="TC:" THEN
+        DO
+         BNOTEXT=SEGMENT(CFGFILELINE,4,LENGTH(CFGFILELINE))
+         IF LENGTH(BNOTEXT)>0 THEN
+          CALL BNONLTS
+         TARGETCOMMENT=BNOTEXT
+        END          
+       IF UPPERCASE(SEGMENT(CFGFILELINE,1,3))="XA:" THEN
+        DO
+         BNOTEXT=SEGMENT(CFGFILELINE,4,LENGTH(CFGFILELINE))
+         IF LENGTH(BNOTEXT)>0 THEN
+          CALL BNONLTS
+         IF CHARACTERSEARCH("YESTRUEON",UPPERCASE(BNOTEXT))>0 THEN
+          XACCTACCESS=TRUE
+        END  
+      END [IF CFGFILEERROR=""...]
+     ELSE IF CFGFILEERROR<>"EOF" THEN
+      DO
+       PRINT "Error reading Letter file '"+CFGFILENAME+"': "+CFGFILEERROR
+       NEWLINE
+       PRINT "TERMINATING"
+       NEWLINE
+       TERMINATE
+      END
+    END [WHILE CFGFILEERROR=""]
+  END [IF CFGFILEERROR=""]
+ ELSE
+  DO
+   PRINT "Letter file '"+CFGFILENAME+"' open error:"+CFGFILEERROR
+   NEWLINE
+   PRINT "TERMINATING"
+   NEWLINE
+   TERMINATE
+  END
+
+[* Print parameters for this run to the Batch Output report
+*]
+ PRINT FORMAT("TARGETSYMXUSER:9999",TARGETSYMXUSER)
+ NEWLINE
+ PRINT FORMAT("TARGETSTARTDATE:99/99/99",TARGETSTARTDATE)
+ NEWLINE
+ PRINT FORMAT("TARGETENDDATE:99/99/99",TARGETENDDATE)
+ NEWLINE
+ PRINT "TARGETCOMMENT:'"+TARGETCOMMENT+"'"
+ NEWLINE
+ PRINT "XACCTACCESS: "+BNOTRUEFALSECHR(XACCTACCESS)
+ NEWLINE
+ NEWLINE
+
+ IF TARGETCOMMENT="" THEN
+  DO
+   PRINT "Target Comment not determined - TERMINATING PROGRAM RUN."
+   NEWLINE
+   TERMINATE
+  END
+END [PROCEDURE]
+
+PROCEDURE GETTRANSACTIONS
+[* Target SymX transactions (by user number)
+** Load a transaction sequence:
+**   A transaction followed by one or more comments
+** Test the transaction:
+**   Either the second comment after the transaction will have the
+**   TARGETCOMMENT as part of the transaction comment field, OR
+**   the TARGETCOMMENT will be part of the transaction
+**   statement description 1 field.
+*]
```

Comment:
I recommend calling CLEARTRANVARS Before the Share Loop, In case somehow a transaction was incomplete and didn't call the procedure.

Reply:
Done